src/components/Article.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from "react";
+import { getArticles, getArticlesById } from "../api";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+import { Comments } from "./Comments";
+
+export const Article = () => {
+  const params = useParams();
+  const [article, setArticle] = useState(null);
+  const [loadingArticle, setLoadingArticle] = useState(false);
+
+  useEffect(() => {
+    setLoadingArticle(true);
+    getArticlesById(params.article_id).then((data) => {
+      setArticle(data.articles);
+      setLoadingArticle(false);
+    });
+  }, [params.article_id]);
```

Comment:
I'm not sure why we would need this in the dependency array? As we are routing to a new page for each article rather than re rendering the same page.

Reply:
I thought that when the article Id changes, it should run useEffect again, I removed it and tried it again and works also. So I removed the dependency and left an empty square bracket.